tests/test_solver.cpp

```diff
@@ -571,3 +571,102 @@ TEST_CASE("amgcl_blocksolver_crystm03_Bicgstab", "[solver]")
     REQUIRE((A * x_b - b).norm() / b.norm() < 1e-7);
 }
 #endif
+
+#ifdef POLYSOLVE_WITH_CUSOLVER
+TEST_CASE("cusolverdn", "[solver]")
+{
+    const std::string path = POLYSOLVE_DATA_DIR;
+    Eigen::SparseMatrix<double> A;
+    const bool ok = loadMarket(A, path + "/A_2.mat");
+    REQUIRE(ok);
+
+    auto solver = LinearSolver::create("cuSolverDN", "");
+    // solver->setParameters(params);
+    Eigen::VectorXd b(A.rows());
+    b.setRandom();
+    Eigen::VectorXd x(b.size());
+    x.setZero();
+
+    solver->analyzePattern(A, A.rows());
+    solver->factorize(A);
+    solver->solve(b, x);
+
+    // std::cout<<"Solver error: "<<x<<std::endl;
+    const double err = (A * x - b).norm();
+    REQUIRE(err < 1e-8);
+}
+TEST_CASE("cusolverdn_dense", "[solver]")
+{
+    const std::string path = POLYSOLVE_DATA_DIR;
+
+    Eigen::MatrixXd A(4, 4);
+    for(int i = 0; i < 4; i++){
+        A(i,i) = 1.0;
+    }
+    A(0,1) = 1.0;
+    A(3,0) = 1.0;
+
+    auto solver = LinearSolver::create("cuSolverDN", "");
+    // solver->setParameters(params);
+    Eigen::VectorXd b(A.rows());
+    b.setRandom();
+    Eigen::VectorXd x(b.size());
+    x.setZero();
+
+    solver->analyzePattern(A, A.rows());
+    solver->factorize(A);
+    solver->solve(b, x);
+
+    // std::cout<<"Solver error: "<<x<<std::endl;
+    const double err = (A * x - b).norm();
+    REQUIRE(err < 1e-8);
+}
+/*
+TEST_CASE("cusolverdn_5cubes", "[solver]")
```

Comment:
why is this off?

Reply:
it requires additional data that isn't included in the polysolve data folder (the gradients and hessians from the 5cubes simulation on polyfem), so i commented it out

Reply:
i can push the data to the repo, so we can have it